File: trees/print.py
class TreeNode(object):
    def __init__(self, data=None, left=None, right=None):
        self.data = data
        self.left = left
        self.right = right

    @classmethod
    def print_inorder(cls, node):
        if node:
            cls.print_inorder(node.left)
            print " {} ".format(node.data),
            cls.print_inorder(node.right)

    @classmethod
    def print_preorder(cls, node):
        if node:
            print " {} ".format(node.data),
            cls.print_preorder(node.left)
            cls.print_preorder(node.right)

    @classmethod
    def print_postorder(cls, node):
        if node:
            cls.print_postorder(node.left)
            cls.print_postorder(node.right)
            print " {} ".format(node.data),


if __name__ == "__main__":
    print """
          Tree
            1
          /   \\
         2     4
        / \\
       3   5
    """

    root = TreeNode(1,
            left=TreeNode(2,
                left=TreeNode(3),
                right=TreeNode(5)),
            right=TreeNode(4))

    print "Inorder:"
    TreeNode.print_inorder(root)

    print
    print "Preorder:"
    TreeNode.print_preorder(root)

    print
    print "Postorder:"
    TreeNode.print_preorder(root)

File: trees/print.c
#include <stdio.h>
#include <stdlib.h>

struct node {
    int data;
    struct node * left;
    struct node * right;
};

struct node * new(int data) {
    struct node * node = (struct node *) malloc(sizeof(struct node));

    node->data = data;
    node->left = NULL;
    node->right = NULL;

    return node;
}

void printPostorder(struct node * node) {
    if (node == NULL)
        return;

    printPostorder(node->left);
    printPostorder(node->right);

    printf("%d ", node->data);
}

void printInorder(struct node * node) {
    if (node == NULL)
        return;

    printInorder(node->left);
    printf("%d ", node->data);
    printInorder(node->right);
}

void printPreorder(struct node * node) {
    if (node == NULL)
        return;

    printf("%d ", node->data);

    printPreorder(node->left);
    printPreorder(node->right);
}

int main() {
    struct node * root = new(1);
    root->left = new(2);
    root->right = new(3);

    root->left->left = new(4);
    root->left->right = new(5);

    printf("\nPreorder traversal of binary tree is \n");
    printPreorder(root);

    printf("\nInorder traversal of binary tree is \n");
    printInorder(root);

    printf("\nPostorder traversal of binary tree is \n");
    printPostorder(root);
}
